composeApp/src/androidMain/kotlin/ru/alexpanov/spacex/common/SlotModalBottomSheetState.kt

```diff
@@ -44,13 +44,15 @@ fun <C : Any, T : Any> rememberSlotModalBottomSheetState(
     val sheetState = rememberModalBottomSheetState(
         initialValue = ModalBottomSheetValue.Hidden,
         skipHalfExpanded = skipHalfExpanded,
-        confirmValueChange = { state ->
-            if (state == ModalBottomSheetValue.Hidden) {
-                onDismiss()
-            }
+        confirmValueChange = {
             true
         }
     )
+    LaunchedEffect(sheetState.targetValue) {
```

Comment:
Thanks a lot for the PR, but there is a one problem with this implementation of the onDismiss callback for the bottom sheet. If you try to not completely hide the bottom sheet without releasing your finger, the onDismiss callback will be called, despite the fact that the bottom sheet is still visible on the screen, thus resulting in inconsistency with the state.

I guess there is a better solution:
```kotlin
    LaunchedEffect(sheetState) {
        snapshotFlow { sheetState.isVisible }
            .distinctUntilChanged()
            .drop(1)
            .collect { visible ->
                if (visible.not()) {
                    onDismiss()
                }
            }
    }
```

Reply:
Indeed. Made edit